src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -24,14 +24,14 @@ static inline void NodeByLabelScanToString(const OpBase *ctx, sds *buf) {
 	ScanToString(ctx, buf, op->n.alias, op->n.label);
```

Comment:
Note, ` NodeByLabelScanConsumeFromChild(OpBase *opBase) {` seems to be calling `_update_label_id`
not sure if it is necessary, also I think the logic there is faulty, please dig into it

Reply:
We added this in #2709 so if we go on with this PR I'll revert this to be up to date. Your idea a lot seems more elegant and simple so I guess we won't..
We removed all other 'run-time' label-id updates, but removing it from `NodeByLabelScanConsumeFromChild` caused several scenarios in Merge1.feature to fall (specifically I remember scenario [9] fell).